### PR TITLE
[CARBONDATA-1910]do not allow tupleid, referenceid and positionReference as columns names

### PR DESCRIPTION
**Pull request by akashrn5**
Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [X] Any interfaces changed?
 NA
 - [X] Any backward compatibility impacted?
 NA
 - [X] Document update required?
NA
 - [X] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
UTs are added to test the scenarios
       
 - [X] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2133/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2422/

**Comment by akashrn5** [comment]
retest this please

**Comment by akashrn5** [comment]
retest sdv please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/929/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2158/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2443/

**Comment by sraghunandan** [comment]
LGTM

File: integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala [hunk 1]
```diff
@@ -173,6 +174,18 @@ class CarbonHelperSqlAstBuilder(conf: SQLConf,
 
     // Ensuring whether no duplicate name is used in table definition
     val colNames = cols.map(_.name)
+
+    // do not allow below key words as column name
+    colNames.foreach { col =>
```
Comment (kumarvishal09) [inline integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala]: Please check scenario for CTAS... create table t1 as select sum(column1) as positionReference from table. In this case name of the column in t1 will be positionReference. So in that case also it should not allow to create carbon table.

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/iud/UpdateCarbonTableTestCase.scala [hunk 2]
```diff
@@ -569,6 +569,39 @@ class UpdateCarbonTableTestCase extends QueryTest with BeforeAndAfterAll {
         CarbonCommonConstants.defaultValueIsPersistEnabled)
   }
 
+  test("test create table with tupleid as column name") {
```
Comment (kumarvishal09) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/iud/UpdateCarbonTableTestCase.scala]: Please add testcase of CTAS

Comment (akashrn5) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/iud/UpdateCarbonTableTestCase.scala]: added

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2318/

**Comment by akashrn5** [comment]
retest this please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1102/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2550/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.0, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1106/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2323/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**